README.md

```diff
@@ -1,39 +1,379 @@
-# Testing
-This is intended to provide tools for embedded testing.
+PHiLIP (Primitive Hardware In the Loop Integration Product)
+===========================================================
+[![Build Status](https://travis-ci.org/riot-appstore/PHiLIP.svg?branch=master)](https://travis-ci.org/riot-appstore/PHiLIP)
 
 
-# Setting up the PHiLIP (Primitive Hardware In the Loop Integration Product)
+## Contents
+<a name="c_desc"></a>
+- [ Description ](#desc)
+<a name="c_setup"></a>
+- [ Setup of PHiLIP Environment ](#setup)
+<a name="c_gs_dev"></a>
+- [ Getting Started as a Developer ](#gs_dev)
+<a name="c_gs_ci"></a>
+- [ Getting Started with CI Scripts ](#gs_ci)
+<a name="c_arch"></a>
+- [ PHiLIP Architecture ](#arch)
+<a name="c_qual"></a>
+- [Qualification of PHiLIP](#qual)
+<a name="c_map"></a>
+- [PHiLIP Memory Map](#map)
+<a name="c_proto"></a>
+- [PHiLIP Serial Protocol](#proto)
+<a name="c_inter"></a>
+- [PHiLIP Python Interface](#inter)
+<a name="c_pinb"></a>
+- [PHiLIP-b Pinout](#pinb)
+<a name="c_pinn"></a>
+- [PHiLIP-n Pinout](#pinn)
+- [PHiLIP Firmware Documentation](https://mrkevinweiss.github.io/philip/doxygen/index.html)
 
-## Tools
-- SWD flasher
-- Soldering iron and solder
+<a name="desc"></a>
+## [Description](#c_desc)
 
-## Materials
-- [bluepill](https://hackaday.com/2017/03/30/the-2-32-bit-arduino-with-debugging/) or [nucelo-f103rb](https://www.digikey.com/products/en?keywords=nucleo-f103rb)
+PHiLIP is qualified open-source firmware for nucleo-f103rb or bluepill boards used for testing peripherals of other embedded devices. PHiLIP is a low-cost solution to allow detailed, corner case peripheral testing for both developers and CI systems. PHiLIP is aimed at getting salient information that would be gathered from an oscilloscope or logic analyzer as well as injecting specific peripheral behaviors. PHiLIP is designed for testing peripheral APIs for embedded operating systems and hardware abstraction layers but was built with an architecture that allows for _easy_ extensions to other applications such as product qualification or simulation. PHiLIP can be used with a raw serial connection but also comes with a python interface that simplifies writing test scripts as well as a shell for developers to run manual tests.
```

Comment:
same as above: break lines to fit within 80-120 chars. Shorter lines typically also work better when using GIT to follow changes (e.g. corrected typos).

applies through this file

Reply:
Ya, I was debating whether I just depend on line wrapping or not.  Lots of things about the project don't fit GIT that well (code generations and csv files seem to not be that great).

Reply:
Happy middle ground of newline per sentence.